### PR TITLE
Wildfly logging receiver

### PR DESCRIPTION
**Pull request by dehaansa**
image

Named wildfly_server as there are wildfly audit logs that we are not targeting for this effort according to the agreed upon spec, but I wanted to leave room for their addition in the future without feeling like a rename would be appropriate.

### CODE REVIEW

**Comment by dehaansa** [comment]
Current debian failures include tomcat & mysql, wildfly is passing.
```--- FAIL: TestThirdPartyApps (0.00s)
    --- FAIL: TestThirdPartyApps/debian-10 (0.00s)
        --- PASS: TestThirdPartyApps/debian-10/elasticsearch (292.62s)
        --- PASS: TestThirdPartyApps/debian-10/redis (349.54s)
        --- PASS: TestThirdPartyApps/debian-10/couchdb (369.34s)
        --- PASS: TestThirdPartyApps/debian-10/postgresql (383.98s)
        --- PASS: TestThirdPartyApps/debian-10/wildfly (425.24s)
        --- PASS: TestThirdPartyApps/debian-10/memcached (435.71s)
        --- PASS: TestThirdPartyApps/debian-10/apache (454.81s)
        --- PASS: TestThirdPartyApps/debian-10/mongodb (456.58s)
        --- PASS: TestThirdPartyApps/debian-10/jvm (489.18s)
        --- PASS: TestThirdPartyApps/debian-10/cassandra (492.70s)
        --- PASS: TestThirdPartyApps/debian-10/nginx (497.46s)
        --- PASS: TestThirdPartyApps/debian-10/solr (501.21s)
        --- FAIL: TestThirdPartyApps/debian-10/mysql (989.27s)
        --- FAIL: TestThirdPartyApps/debian-10/tomcat (1004.48s)```

**Comment by qingling128** [comment]
Adding @ridwanmsharif for reverse shadow.

File: apps/wildfly.go
Comment (ridwanmsharif) [inline apps/wildfly.go:56]: Is there a reason its
`\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}.\d{3,}` instead of
`\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2},\d{3}`?

Looks like in all the cases above the secs and nanosecs are separated by a `,` and the nanosecs are 3 digits

File: apps/wildfly.go
Comment (ridwanmsharif) [inline apps/wildfly.go:93]: https://docs.wildfly.org/26/Admin_Guide.html#default-log-file-locations

There seem to be a couple more log locations, namely `./domain/log/*.log` for the host and process controller according to the above link. Are they out of scope?

File: apps/wildfly.go
Comment (ridwanmsharif) [inline apps/wildfly.go:27]: Looks like we're naming this because we're not targeting [Audit logs ](https://docs.wildfly.org/26/Admin_Guide.html#Audit_logging). Looks like audit logs are off by default, and have a different structure. Not too sure what naming pattern we should use here. I was thinking maybe having this be `wildfly` and for audit logs we'd have `wildfly_audit`. Will defer to @qingling128's judgement

File: docs/wildfly.md
Comment (ridwanmsharif) [inline docs/wildfly.md:10]: [nit] `s/apache/wildfly`

File: integration_test/third_party_apps_data/agent/ops-agent/linux/enable_wildfly
Comment (ridwanmsharif) [inline integration_test/third_party_apps_data/agent/ops-agent/linux/enable_wildfly:31]: Do we need to add in a sleep here? Looks like we do it for some of the other applications

**Review by ridwanmsharif — COMMENTED** [review]
Left a few comments over my first scan

File: apps/wildfly.go
Comment (dehaansa) [reply apps/wildfly.go:56]: Great catch, that should be a comma. This was copied from another app's start state and clearly wasn't fixed.

**Comment by dehaansa** [comment]
As of commit https://github.com/GoogleCloudPlatform/ops-agent/pull/382/commits/bab6dcbb5cb42ecb4dd22b2e4fdd4886b1b567e1 all integration tests passing other than known postgresql issues.

Outstanding concerns:
Investigating domain controller logs to see if they are formatted the same and can be dropped in without issue.
Waiting @qingling128 's perspective on naming comment https://github.com/GoogleCloudPlatform/ops-agent/pull/382#discussion_r803180337

**Review by qingling128 — APPROVED** [review]
LGTM module @ridwanmsharif 's feedback is addressed. I also left some minor comments.

@ridwanmsharif Feel free to merge if the comments are addressed and tests passed. (aka I will check back periodically, too. But another round of reviews by me is not required, as the feedback is minor.)

File: apps/wildfly.go
Comment (dehaansa) [reply apps/wildfly.go:27]: I'm interpreting @qingling128 's comments to rename pipelines to "wildfly_system" as an acceptance of the current naming. Confirm here if that wasn't your intention Ling!

File: integration_test/third_party_apps_data/applications/wildfly/centos_rhel/install
Comment (dehaansa) [reply integration_test/third_party_apps_data/applications/wildfly/centos_rhel/install:7]: Left the install scripts in place in case this changes at some point in the future, but used the common_skips to pass on  non-debian 10.

File: apps/wildfly.go
Comment (qingling128) [reply apps/wildfly.go:27]: Either `wildfly_server` or `wildfly_system` is OK to me.

I do see `server logs` pop up in their documentation here and there. And the current log files say `server.log`.

If we are adding `./domain/log/*.log` though, it's better as `wildfly_system` logs so that it's more generic.

File: apps/wildfly.go
Comment (ridwanmsharif) [reply apps/wildfly.go:75]: Have we verified that the logs are formatted differently for these additional files? As per the naming discussion, we would rename the receiver to `wildfly_server` if those logs are being added.

**Review by ridwanmsharif — APPROVED** [review]
LGTM. Will merge after we decide on the `./domain/log/*.log` files

**Comment by dehaansa** [comment]
Re: https://github.com/GoogleCloudPlatform/ops-agent/pull/382#discussion_r804800902 - hoping to have this well understood early Monday, so this'll have to wait until then.

File: apps/wildfly.go
Comment (dehaansa) [reply apps/wildfly.go:77]: Confirmed that controller logs are the same format, updated.

**Pull request merged**
